src/java.desktop/windows/native/libjsound/PLATFORM_API_WinOS_DirectSound.cpp

```diff
@@ -224,6 +224,7 @@ INT32 DAUDIO_GetDirectAudioDeviceCount() {
 
         g_lastCacheRefreshTime = (UINT64) timeGetTime();
     }
+    ::CoUninitialize();
```

Comment:
I guess this should be called only if the CoInitialize returns S_OK or S_FALSE. It is also necessary to check what happens when the user(via JNI or any other API) initialized COM before using different "params" - I assume in this case CoInitialize in our code should return S_FALSE.

Reply:
Thank you for the review, added status check.

Reply:
I think it is better init com before the loc, and disable after. Note that the DAUDIO_GetDirectAudioDeviceCount() has similar issue.